_projects/result-reporter-tool.md

```diff
@@ -0,0 +1,86 @@
+---
+collaborating_projects:
+  - coala
+desc: "Implement a result reporter tool that converts linter/bear results
+into various formats"
+difficulty: medium
+initiatives:
+  - GSoC
+issues:
+  - "https://github.com/coala/projects/issues/555"
+  - "https://github.com/coala/coala/issues/2757"
+  - "https://github.com/coala/coala/issues/5910"
+  - "https://gitlab.com/coala/coala-json/issues/3"
+markdown: result-reporter-tool.md
+mentors:
+  - li-boxuan
+name: "Result Reporter Tool"
+requirements:
+  - "The applicant should have at least one non-trivial patch merged to
+    coala or coala-bears."
+  - "The applicant should be familiar with Python."
+  - "Familiarity with continuous integration tools (e.g. Jenkins) is a plus."
+tags:
+  - Core
+  - Linter
+  - CI
+---
+
+Apart from common CLI output, coala provides json format output. However, there
+are other formats which are useful for reporting tests and failures, but coala
+does not support at the moment.
+
+This is not only for coala, result format inconsistencies has been a problem for
+a long time. Converting the static analysis results into a test results format
+has been done a few times, such as early PEP8 plugins to Jenkins. The mapping
+isn't exact, but the benefits of using the test result format is tight
+integration with various systems.
+
+Most commonly used is JUnit format. CircleCI can accept
+[this format](https://circleci.com/docs/2.0/collect-test-data). AppVeyor can
+receive test results in [XML format](https://www.appveyor.com/docs/running-tests/#uploading-xml-test-results).
+Other formats include [TAP](https://testanything.org/),
+[Clover](http://openclover.org/doc/manual/latest/ant--clover-report.html),
+There are some systems, such as Phabricator, which have a
+[custom format](https://github.com/facebook/infer/issues/673) they expect.
+
+The applicant is welcomed to propose additional ideas around format topic, to
+improve coala workflow, continuous integration workflow or linters workflow.
+For example, pulling the coala results from CI could be a way to inform newbies
+of the style problems with their PRs. Pushing the unit test results as XML to
```

Comment:
Can you please explain, In what way would pushing the unit test results as XML to the CI help?

I mean, Isn't our CI run when someone makes a PR. And whenever  CI is run all the unit tests are executed by default and the results can be seen. Sorry if this might sound silly, but I haven't yet come across a case when we had to push the XML's of unit tests.

Reply:
https://circleci.com/docs/2.0/collect-test-data/ is a good example